src/dataflow/src/source/kafka.rs

```diff
@@ -682,7 +682,18 @@ where
             // via individual partition queues.
             {
                 let message = dp_info.consumer.poll(Duration::from_secs(0));
-                assert!(message.is_none());
+                if !message.is_none() {
+                    match message.unwrap() {
+                        Ok(message) => {
+                            panic!(
+                                "Received an unexpected message Offset: {} PID: {} on main partition loop",
+                                message.partition(),
```

Comment:
shouldn't line 690 and line 691 be switched around?

Reply:
We should panic when we receive a message (as we should never receive messages on the main consumer loop), but we can possibly receive errors.

Reply:
Oh sorry, I see what you mean. Yes, offset/partition should be switched.